architecture-compose/src/androidLibMain/kotlin/com/splendo/kaluga/architecture/compose/navigation/Route.kt

```diff
@@ -0,0 +1,205 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.architecture.compose.navigation
+
+import com.splendo.kaluga.architecture.navigation.NavigationAction
+import com.splendo.kaluga.architecture.navigation.NavigationBundleSpec
+import com.splendo.kaluga.architecture.navigation.NavigationBundleSpecRow
+import com.splendo.kaluga.architecture.navigation.NavigationBundleSpecType
+import com.splendo.kaluga.architecture.navigation.NavigationBundleValue
+import com.splendo.kaluga.base.text.DateFormatter
+import com.splendo.kaluga.base.text.iso8601Pattern
+import kotlinx.serialization.builtins.BooleanArraySerializer
+import kotlinx.serialization.builtins.ByteArraySerializer
+import kotlinx.serialization.builtins.CharArraySerializer
+import kotlinx.serialization.builtins.DoubleArraySerializer
+import kotlinx.serialization.builtins.FloatArraySerializer
+import kotlinx.serialization.builtins.IntArraySerializer
+import kotlinx.serialization.builtins.ListSerializer
+import kotlinx.serialization.builtins.LongArraySerializer
+import kotlinx.serialization.builtins.ShortArraySerializer
+import kotlinx.serialization.builtins.serializer
+import kotlinx.serialization.json.Json
+import kotlin.reflect.KClass
+
+/** Reserved route name for the back button navigation. */
+internal const val ROOT_VIEW = "com.splendo.kaluga.architecture.compose.navigation.root_view"
+
+val NavigationBundleSpecRow<*>.argumentKey: String get() = key ?: javaClass.simpleName
+
+/** @return a route represented by the [NavigationAction]. */
+inline fun <SpecRow : NavigationBundleSpecRow<*>, reified Action : NavigationAction<SpecRow>> route(
+    spec: NavigationBundleSpec<SpecRow>
+): String = route(Action::class, spec)
+
+fun <SpecRow : NavigationBundleSpecRow<*>, T : NavigationAction<SpecRow>> route(
+    actionClass: KClass<T>,
+    spec: NavigationBundleSpec<SpecRow>
+): String {
+    val arguments = spec.rows.mapNotNull { row ->
+        val key = row.argumentKey
+        when (row.associatedType) {
+            is NavigationBundleSpecType.UnitType -> null
+            is NavigationBundleSpecType.OptionalType<*> -> Triple(key, "{$key}", false)
```

Comment:
using a data class instead of `Triple` would allow to name the arguments providing more context

Reply:
Done